PactNet/Mocks/MockHttpService/MockProviderService.cs

```diff
@@ -30,9 +30,9 @@ public class MockProviderService : IMockProviderService
             _adminHttpClient = adminHttpClientFactory(BaseUri);
         }
 
-        public MockProviderService(int port, bool enableSsl, string consumerName, string providerName, PactConfig config)
+        public MockProviderService(int port, bool enableSsl, string consumerName, string providerName, PactConfig config,string host = "")
             : this(
-            baseUri => new RubyHttpHost(baseUri, consumerName, providerName, config),
+            baseUri => new RubyHttpHost(baseUri, consumerName, providerName, config,host),

```

Comment:
RubyHttpHost should not need to take in the host, you should update the baseUri to reflect the host that is being used.

This is done by passing host to the MockProviderService internal ctor and changing `BaseUri = new Uri($"{(enableSsl ? "https" : "http")}://localhost:{port}");`